File: python/trie_hash/Solution128.py
# 128. 最长连续序列
from typing import List


class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        tool = set(nums)
        longest = 0
        for num in nums:
            if (num - 1) not in tool:
                cur_num = num
                cur_length = 1
                while (cur_num + 1) in tool:
                    cur_num += 1
                    cur_length += 1
                longest = max(longest, cur_length)
        return longest

File: src/trie_hash/Solution128.java
// 128. 最长连续序列

package trie_hash;

import java.util.*;

public class Solution128 {
    public int longestConsecutive(int[] nums) {
        if (nums == null || nums.length == 0) return 0;
        Set<Integer> set = new HashSet<>();
        // build the set
        for (int num: nums) {
            set.add(num);
        }

        int maxLength = 1;
        for (int num: nums) {
            if (!set.contains(num - 1)) {
                int curLength = 1;
                while (set.contains(num + 1)) {
                    num++;
                    curLength++;
                    maxLength = Math.max(maxLength, curLength);
                }
            }
        }
        return maxLength;
    }
}
